cargo-afl/src/bin/cargo-afl.rs

```diff
@@ -297,18 +299,49 @@ where
 
     // `-C codegen-units=1` is needed to work around link errors
     // https://github.com/rust-fuzz/afl.rs/pull/193#issuecomment-933550430
-    let mut rustflags = format!(
-        "-C debug-assertions \
-         -C overflow_checks \
-         -C passes={passes} \
-         -C codegen-units=1 \
-         -C llvm-args=-sanitizer-coverage-level=3 \
-         -C llvm-args=-sanitizer-coverage-trace-pc-guard \
-         -C llvm-args=-sanitizer-coverage-prune-blocks=0 \
-         -C llvm-args=-sanitizer-coverage-trace-compares \
-         -C opt-level=3 \
-         -C target-cpu=native "
-    );
+
+    let binding = common::afl_llvm_dir(None);
+    let p = binding.display();
+
+    let mut rustflags = String::new();
+
+    if cfg!(feature = "cmplog") {
+        // Make sure we are on nightly for the -Z flags
+        assert!(
+            rustc_version::version_meta().unwrap().channel == rustc_version::Channel::Nightly,
+            "cargo-afl must be compiled with nightly for the cmplog feature"
+        );
+
+        let llvm_config = common::get_llvm_config();
+
+        // check if llvm tools are installed and with the good version for the plugin compilation
+        let mut command = Command::new(llvm_config.clone());
+        command.args(["--version"]);
+        let status = command
+            .status()
+            .unwrap_or_else(|_| panic!("could not run {llvm_config} --version"));
+        assert!(status.success());
+
+        rustflags = format!(
+            "-C debug-assertions \
+             -C overflow_checks \
+             -C passes={passes} \
+             -C codegen-units=1 \
+             -Z llvm-plugins={p}/cmplog-instructions-pass.so  \
+             -Z llvm-plugins={p}/cmplog-routines-pass.so \
+             -Z llvm-plugins={p}/cmplog-switches-pass.so \
+             -Z llvm-plugins={p}/SanitizerCoveragePCGUARD.so \
+             -C opt-level=3 \
+             -C target-cpu=native "
+        );
+    } else {
+        rustflags = "-C llvm-args=-sanitizer-coverage-level=3 \
+           -C llvm-args=-sanitizer-coverage-trace-pc-guard \
+           -C llvm-args=-sanitizer-coverage-prune-blocks=0 \
+           -C llvm-args=-sanitizer-coverage-trace-compares
+           "
+        .to_string();
+    }
```

Comment:
I think the else branch is missing some flags.

I think it should be okay to reorder the flags. I would set more of them prior to entering this `if` condition, and then use the `if` condition to only set the ones that differ when `cmplog` is set/unset.